<!DOCTYPE html>
<html lang="en"><head>
<link href="../assets/SIB_logo.svg" rel="icon" type="image/svg+xml">
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-597958c53c93a607afca12fd375c57ed.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.26">

  <meta name="author" content="Flavio Lombardo">
  <meta name="dcterms.date" content="2025-12-12">
  <title>Cancer variant analysis – Cancer Variant Analysis</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
      }
    pre.numberSource { margin-left: 3em;  padding-left: 4px; }
    div.sourceCode
      { color: #24292e;  }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #24292e; } /* Normal */
    code span.al { color: #ff5555; font-weight: bold; } /* Alert */
    code span.an { color: #6a737d; } /* Annotation */
    code span.at { color: #d73a49; } /* Attribute */
    code span.bn { color: #005cc5; } /* BaseN */
    code span.bu { color: #d73a49; } /* BuiltIn */
    code span.cf { color: #d73a49; } /* ControlFlow */
    code span.ch { color: #032f62; } /* Char */
    code span.cn { color: #005cc5; } /* Constant */
    code span.co { color: #6a737d; } /* Comment */
    code span.cv { color: #6a737d; } /* CommentVar */
    code span.do { color: #6a737d; } /* Documentation */
    code span.dt { color: #d73a49; } /* DataType */
    code span.dv { color: #005cc5; } /* DecVal */
    code span.er { color: #ff5555; text-decoration: underline; } /* Error */
    code span.ex { color: #d73a49; font-weight: bold; } /* Extension */
    code span.fl { color: #005cc5; } /* Float */
    code span.fu { color: #6f42c1; } /* Function */
    code span.im { color: #032f62; } /* Import */
    code span.in { color: #6a737d; } /* Information */
    code span.kw { color: #d73a49; } /* Keyword */
    code span.op { color: #24292e; } /* Operator */
    code span.ot { color: #6f42c1; } /* Other */
    code span.pp { color: #d73a49; } /* Preprocessor */
    code span.re { color: #6a737d; } /* RegionMarker */
    code span.sc { color: #005cc5; } /* SpecialChar */
    code span.ss { color: #032f62; } /* SpecialString */
    code span.st { color: #032f62; } /* String */
    code span.va { color: #e36209; } /* Variable */
    code span.vs { color: #032f62; } /* VerbatimString */
    code span.wa { color: #ff5555; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8e6b04b590b725f58e44cc1786f616e4.css">
  <link rel="stylesheet" href=".reveal .slides {
  font-size: 1.5em;
}
.reveal h2 {
  font-size: 2em;
}
.reveal .callout {
  font-size: 1.5em;
}
.reveal table {
  font-size: 1.2em;
}
.reveal code {
  font-size: 1.2em;
}">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  <script src="../site_libs/quarto-diagram/mermaid.min.js"></script>
  <script src="../site_libs/quarto-diagram/mermaid-init.js"></script>
  <link href="../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Cancer Variant Analysis</h1>
  <p class="subtitle">Introduction to Cancer Genomics and Variant Detection</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Flavio Lombardo 
</div>
</div>
</div>

  <p class="date">2025-12-12</p>
</section>
<section id="learning-objectives" class="slide level2">
<h2>Learning Objectives</h2>
<p>After this session, you will be able to:</p>
<ul>
<li>Understand why cancer is fundamentally a genomic disease</li>
<li>Distinguish between somatic and germline variants</li>
<li>Identify different types of mutations in cancer</li>
<li>Understand sequencing approaches for cancer genomics</li>
<li>Comprehend the bioinformatics workflow for variant detection</li>
<li>Appreciate the clinical relevance of variant calling</li>
</ul>
<aside class="notes">
<p>Start by setting expectations. Cancer genomics is a rapidly evolving field that combines molecular biology with computational analysis.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="cancer-a-disease-of-the-genome" class="slide level2">
<h2>Cancer: A Disease of the Genome</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><strong>Key characteristics:</strong></p>
<ul>
<li><strong>Abnormal cell growth</strong> - uncontrolled proliferation</li>
<li><strong>Invasive potential</strong> - ability to spread (metastasis)</li>
<li>Initiated by <strong>acquired genomic mutations</strong> affecting cell growth regulators</li>
<li>Mutations occur <strong>stochastically</strong>, but rate influenced by environment</li>
<li><strong>Clonal evolution</strong> - natural selection of malignant cells</li>
</ul>
</div><div class="column" style="width:40%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/01_lung_metastases_CT_image.png" alt="CT scan showing liver-metastatic cancer"></p>
<figcaption>CT image showing multiple lung masses in a patient with metastatic testicular seminoma, <a href="https://doi.org/10.1102/1470-7330.2006.0020">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Driver vs Passenger Mutations</strong></p>
</div>
<div class="callout-content">
<p>Cancer genomes typically harbor <strong>2-8 driver mutations</strong> (Vogelstein et al., 2013), though this varies by cancer type. Pediatric cancers often have fewer drivers, while hypermutated adult cancers may have more. Drivers represent a small minority of total mutations.</p>
</div>
</div>
</div>
</section>
<section id="hereditary-vs-sporadic-cancer" class="slide level2">
<h2>Hereditary vs Sporadic Cancer</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="sporadic-90-95">Sporadic (~90-95%)</h3>
<ul>
<li>Acquired somatic mutations</li>
<li>Accumulate during lifetime</li>
<li>Environmental factors contribute</li>
</ul>
<h3 id="hereditary-5-10">Hereditary (~5-10%)</h3>
<ul>
<li>Inherited predisposition alleles</li>
<li><strong>BRCA1/2</strong> (breast, ovarian)</li>
<li><strong>Lynch syndrome</strong> genes (colorectal)</li>
<li>Still require somatic “second hits”</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/02_two_hits_hypot.jpg" alt="Knudson hypothesis"></p>
<figcaption>Knudson hypothesis: Tumor Suppressor Gene, <a href="https://josefelixrodriguezantonweb.com/tag/two-hit-hypothesis/">Source</a></figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/02b_germline_vs_somatic.jpg" alt="Somatic mutations create masaicism"></p>
<figcaption>Somatic mutations, <a href="https://doi.org/10.1016/j.ymeth.2019.11.002">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Clinical Relevance</strong></p>
</div>
<div class="callout-content">
<p>Hereditary cancer syndromes have implications for family screening, risk reduction, and treatment options (e.g., PARP inhibitors for BRCA carriers).</p>
</div>
</div>
</div>
</section>
<section id="the-hallmarks-of-cancer" class="slide level2">
<h2>The Hallmarks of Cancer</h2>
<p><strong>8 Core Hallmarks:</strong></p>
<table class="caption-top">
<colgroup>
<col style="width: 43%">
<col style="width: 56%">
</colgroup>
<thead>
<tr class="header">
<th>Hallmark</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Sustaining proliferative signaling</td>
<td>Self-sufficiency in growth signals</td>
</tr>
<tr class="even">
<td>Evading growth suppressors</td>
<td>Insensitivity to anti-growth signals</td>
</tr>
<tr class="odd">
<td>Resisting cell death</td>
<td>Evading apoptosis</td>
</tr>
<tr class="even">
<td>Enabling replicative immortality</td>
<td>Limitless replicative potential</td>
</tr>
<tr class="odd">
<td>Inducing angiogenesis</td>
<td>Sustained blood vessel formation</td>
</tr>
<tr class="even">
<td>Activating invasion &amp; metastasis</td>
<td>Tissue invasion and spread</td>
</tr>
<tr class="odd">
<td>Avoiding immune destruction</td>
<td>Escaping immune surveillance</td>
</tr>
<tr class="even">
<td>Deregulating cellular energetics</td>
<td>Altered metabolism (Warburg effect)</td>
</tr>
</tbody>
</table>
<p><strong>2 Enabling Characteristics:</strong> Genome instability &amp; mutation; Tumor-promoting inflammation</p>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Reference</strong></p>
</div>
<div class="callout-content">
<p>Hanahan &amp; Weinberg (2011) “Hallmarks of Cancer: The Next Generation” - Cell</p>
</div>
</div>
</div>
</section>
<section id="variant-terminology-essential-definitions" class="slide level2">
<h2>Variant Terminology: Essential Definitions</h2>
<div class="columns">
<div class="column" style="width:55%;">
<p><strong>Variant</strong>: A difference in DNA sequence compared to a reference</p>
<p><strong>Mutation</strong>: The <em>process</em> of change in DNA</p>
<p><strong>Somatic variant</strong>: Occurs only in specific cells/tissues - Not inherited - Arises during lifetime</p>
<p><strong>Germline variant</strong>: Present in all cells - Can be passed to offspring - Present from conception</p>
<p><strong>Polymorphism</strong>: Traditionally defined as variant &gt;1% frequency in population (though “variant” is now preferred terminology)</p>
</div><div class="column" style="width:45%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/03_PTC_hereditability.png" alt="Pedigree showing inheritance pattern"></p>
<figcaption>PTC example: pedigree charts, <a href="https://www.bbc.co.uk/bitesize/guides/z2xbh39/revision/5">Source</a></figcaption>
</figure>
</div>
</div></div>
</section>
<section id="mutation-vs-variant-a-practical-example" class="slide level2">
<h2>Mutation vs Variant: A Practical Example</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>The Yellow Flower Example:</strong></p>
<ul>
<li><strong>Mutation</strong>: The change in DNA that caused petals to turn yellow</li>
<li><strong>Variant</strong>: The resulting DNA difference between yellow and white flowers</li>
</ul>
<p>This is a <strong>somatic mutation</strong> - occurring during the plant’s development in one branch, analogous to somatic mutations in cancer.</p>
<p>The CCD4a gene mutation prevents breakdown of carotenoids, leading to yellow pigmentation.</p>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/04_mutations_effects.jpg" style="width:60.0%" alt="Yellow and white chrysanthemum flowers"></p>
<figcaption>Mutation example: chrysanthemum flowers (credit: Geert van Gest). <a href="https://doi.org/10.1104/pp.106.087130">Source</a></figcaption>
</figure>
</div>
</div></div>
</section>
<section id="types-of-mutations-in-cancer" class="slide level2">
<h2>Types of Mutations in Cancer</h2>
<h3 id="small-scale-mutations">Small-scale mutations</h3>
<ul>
<li><strong>SNVs</strong> (Single Nucleotide Variants)</li>
<li><strong>INDELs</strong> (Insertions and Deletions)</li>
</ul>
<h3 id="structural-variations">Structural variations</h3>
<ul>
<li><strong>Large INDELs</strong> (&gt;50bp)</li>
<li><strong>Translocations</strong> - chromosomal rearrangements</li>
<li><strong>Inversions</strong> - reversed DNA segments</li>
<li><strong>Fusion transcripts</strong> - gene fusions from translocations</li>
<li><strong>CNV</strong> (Copy Number Variation) - gains/losses of genomic regions</li>
</ul>
<p><strong>LOH</strong> (Loss of Heterozygosity) is a <em>consequence</em> of CNV or copy-neutral events, not a mutation type itself.</p>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Clinical Relevance</strong></p>
</div>
<div class="callout-content">
<p>Different mutation types require different detection methods and have distinct clinical implications for treatment selection.</p>
</div>
</div>
</div>
</section>
<section id="mutational-signatures-fingerprints-of-mutagenesis" class="slide level2">
<h2>Mutational Signatures: Fingerprints of Mutagenesis</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Each mutational process leaves a characteristic pattern:</p>
<p><strong>Signature 1</strong>: Aging (C&gt;T at CpG)</p>
<p><strong>Signature 4</strong>: Tobacco smoking (C&gt;A)</p>
<p><strong>Signature 7</strong>: UV light (C&gt;T at dipyrimidines)</p>
<p><strong>Signature 6</strong>: Mismatch repair deficiency</p>
<p><strong>Clinical applications:</strong></p>
<ul>
<li>Understanding tumor etiology</li>
<li>Treatment selection (e.g., PARP inhibitors for HRD)</li>
<li>Immunotherapy response prediction</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/05B_SBS1_GRCh38.jpg" alt="Mutational signature spectrum plot, SBS1, associated with ageing"></p>
<figcaption>Mutational signature SBS1: Ageing signature. <a href="https://cancer.sanger.ac.uk/signatures/sbs/sbs4/">Source</a></figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/05_SBS4_GRCh38.jpg" alt="Mutational signature spectrum plot, SBS4, associated with smoking"></p>
<figcaption>Mutational signature SBS4: Smoking signature. <a href="https://cancer.sanger.ac.uk/signatures/sbs/sbs1/">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Reference</strong></p>
</div>
<div class="callout-content">
<p>Alexandrov et al.&nbsp;(2020) “The repertoire of mutational signatures in human cancer” - Nature</p>
</div>
</div>
</div>
</section>
<section id="sequencing-strategies-for-cancer-genomics" class="slide level2">
<h2>Sequencing Strategies for Cancer Genomics</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="coverage-strategies">Coverage strategies</h3>
<ul>
<li><strong>Whole Genome Sequencing (WGS)</strong>
<ul>
<li>Complete genome coverage</li>
<li>Detects all variant types</li>
<li>Best for structural variants</li>
</ul></li>
<li><strong>Whole Exome Sequencing (WES) (Bait Capture)</strong>
<ul>
<li>Protein-coding regions only</li>
<li>Cost-effective (<br>WES 100x: 25 M 2 x 100 bp<br>WGS 30x: 450 M 2 x 100 bp<br>)</li>
<li>Misses non-coding regions</li>
</ul></li>
<li><strong>Custom panels (Bait Capture, mostly)</strong>
<ul>
<li>Targeted cancer genes</li>
<li>Deep coverage</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<h3 id="sequencing-technologies">Sequencing technologies</h3>
<p><strong>Short reads (2x150bp standard):</strong></p>
<ul>
<li>Illumina, MGI, Element, Ultima</li>
<li>High accuracy, mature pipelines</li>
</ul>
<p><strong>Long reads:</strong></p>
<ul>
<li>PacBio HiFi: &gt;Q30 accuracy</li>
<li>Oxford Nanopore: Q20+ with R10.4.1</li>
<li>Superior for SVs and phasing</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/06_sequencing.png" alt="Sequencing technologies"></p>
<figcaption>Sequencing cost vs throughput, credit: Geert van Gest</figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Depth Recommendations</strong></p>
</div>
<div class="callout-content">
<p>WGS: 60-100x tumor, 30-40x normal WES: 100-200x ctDNA: 10,000-30,000x</p>
</div>
</div>
</div>
</section>
<section id="experimental-design-tumor-normal-pairs" class="slide level2 scrollable">
<h2>Experimental Design: Tumor-Normal Pairs</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>The challenge:</strong> Tumor tissue is heterogeneous - contains: - Tumor cells - Immune infiltrates - Stromal cells - Normal tissue</p>
<p><strong>Solution: Paired samples</strong></p>
<ol type="1">
<li><strong>Tumor sample</strong> - from the malignancy</li>
<li><strong>Normal sample</strong> - typically blood</li>
</ol>
<p><strong>For hematological malignancies</strong>, use skin biopsy or buccal swab (blood IS the tumor!)</p>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/07_CELA1.png" alt="IGV showing tumor vs normal comparison"></p>
<figcaption>IGV normal matched vs tumor</figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tumor Purity</strong></p>
</div>
<div class="callout-content">
<p>Typical tumor purity ranges from 30-80%. <strong>Samples below 20-30% purity may have insufficient power for reliable somatic variant detection.</strong> Consider pathologist review or microdissection.</p>
</div>
</div>
</div>
</section>
<section id="bioinformatics-workflow-overview" class="slide level2">
<h2>Bioinformatics Workflow Overview</h2>
<div class="cell" data-reveal="true" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class=""><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
    A[Raw FASTQ] --&gt; B[QC + Adapter&lt;br/&gt;fastp/MultiQC]
    B --&gt; C[Alignment&lt;br/&gt;bwa-mem2]
    C --&gt; D[Add Read Groups]
    D --&gt; E[Mark Duplicates&lt;br/&gt;GATK]
    E --&gt; F[BQSR]
    F --&gt; G[Contamination&lt;br/&gt;Check]
    G --&gt; H[Mutect2]
    H --&gt; I[Filter&lt;br/&gt;Calls]
    I --&gt; J[Annotation&lt;br/&gt;VEP]
</pre>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Critical Note</strong></p>
</div>
<div class="callout-content">
<p>Variant <strong>annotation</strong> is essential but often overlooked! Without functional annotation and clinical database lookup, variants have limited utility.</p>
</div>
</div>
</div>
</section>
<section id="the-sambamcram-file-formats" class="slide level2">
<h2>The SAM/BAM/CRAM File Formats</h2>
<p><strong>Purpose:</strong> Store sequence alignments</p>
<div class="columns">
<div class="column" style="width:40%;">
<p><strong>Key points:</strong></p>
<ul>
<li><strong>SAM</strong>: Text-based (Human readable)</li>
<li><strong>BAM</strong>: <strong>Lossless</strong> binary compressed SAM
<ul>
<li><em>Indexable (Fast random access)</em></li>
</ul></li>
<li><strong>CRAM</strong>: <strong>Ref-based</strong> compression
<ul>
<li><em>Lossless or Lossy modes</em></li>
<li>~30-60% smaller than BAM</li>
<li>Requires access to the reference genome for decoding</li>
</ul></li>
</ul>
</div><div class="column" style="width:60%;">
<p>Example of <code>header</code> lines from a bam file</p>
<pre><code>@HD VN:1.6  SO:coordinate
@SQ SN:chr6 LN:170805979
@SQ SN:chr17    LN:83257441
@RG ID:HWI-ST466.C1TD1ACXX.normal   LB:normal   PL:ILLUMINA SM:normal   PU:HWI-ST466.C1TD1ACXX
@PG ID:bwa  PN:bwa  VN:0.7.17-r1188 CL:bwa mem /config/data/reference//ref_genome.fa /config/data/reads/normal_R1.fastq.gz /config/data/reads/normal_R2.fastq.gz
@PG ID:samtools PN:samtools PP:bwa  VN:1.21 CL:samtools sort
@PG ID:samtools.1   PN:samtools PP:samtools VN:1.21 CL:samtools view -bh</code></pre>
<table class="caption-top">
<colgroup>
<col style="width: 9%">
<col style="width: 90%">
</colgroup>
<thead>
<tr class="header">
<th>Field</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>QNAME</td>
<td>Query (read) name. Identifies each sequencing read.</td>
</tr>
<tr class="even">
<td>FLAG</td>
<td>Bitwise flags describing read state (paired, mapped, strand, etc.).</td>
</tr>
<tr class="odd">
<td>RNAME</td>
<td>Reference sequence (chromosome) the read is aligned to.</td>
</tr>
<tr class="even">
<td>POS</td>
<td>1-based leftmost mapping position on the reference.</td>
</tr>
<tr class="odd">
<td>MAPQ</td>
<td>Mapping quality score indicating confidence of alignment.</td>
</tr>
<tr class="even">
<td>CIGAR</td>
<td>Encodes how the read aligns (matches, insertions, deletions, splicing).</td>
</tr>
<tr class="odd">
<td>SEQ</td>
<td>The nucleotide sequence of the read.</td>
</tr>
<tr class="even">
<td>QUAL</td>
<td>Phred-scaled base quality scores for the read sequence.</td>
</tr>
</tbody>
</table>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Resource</strong></p>
</div>
<div class="callout-content">
<p>Learn more at the SIB course: <a href="https://sib-swiss.github.io/NGS-introduction-training/latest/day2/file_types/">NGS - Quality Control, Alignment, Visualization</a></p>
</div>
</div>
</div>
</section>
<section id="marking-pcr-duplicates" class="slide level2">
<h2>Marking PCR Duplicates</h2>
<div class="columns">
<div class="column" style="width:55%;">
<p><strong>Why it matters:</strong></p>
<ul>
<li>Variant callers assume each read is an <strong>independent observation</strong></li>
<li>PCR/optical duplicates violate this assumption</li>
<li>Can lead to <strong>false positive</strong> variant calls</li>
</ul>
<p><strong>Solution:</strong></p>
<ul>
<li>Mark duplicates based on alignment coordinates</li>
<li>Use <strong>Unique Molecular Identifiers (UMIs)</strong> for accurate deduplication - especially important for low-input and ctDNA samples</li>
</ul>
<p><strong>Tool:</strong> <code>gatk MarkDuplicates</code></p>
</div><div class="column" style="width:45%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/10_mark_duplicates.jpg" alt="Diagram showing PCR duplicate reads"></p>
<figcaption>Mark duplicates, <a href="https://nonacus.com/blog-unique-molecular-identifiers-unmask-low-frequency-variants/">Source</a></figcaption>
</figure>
</div>
</div></div>
</section>
<section id="read-groups-organizing-your-data" class="slide level2">
<h2>Read Groups: Organizing Your Data</h2>
<p><strong>Purpose:</strong> Track metadata for groups of reads within BAM files</p>
<pre><code>@RG ID:rg1 LB:lib1 SM:sampleA PL:ILLUMINA PU:flowcell1.lane1
@RG ID:rg2 LB:lib2 SM:sampleA PL:ILLUMINA PU:flowcell1.lane2

read1  456345 chr20 ... RG:Z:rg1
read2  456348 chr20 ... RG:Z:rg2</code></pre>
<p><strong>Why read groups matter:</strong></p>
<ul>
<li><strong>BQSR</strong> models are built per-read-group</li>
<li><strong>Merging Lanes</strong> Allows combining multiple lanes into one patient file (SM)</li>
<li><strong>Duplicate marking</strong> identifies duplicates within same library</li>
<li>Essential for <strong>QC and troubleshooting</strong></li>
<li>Required for <strong>multi-sample variant calling</strong></li>
</ul>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Best Practice</strong></p>
</div>
<div class="callout-content">
<p>Add read groups during alignment (<code>bwa mem -R</code>) rather than post-hoc (GATK AddOrReplaceReadGroups), for efficiency.</p>
</div>
</div>
</div>
</section>
<section id="somatic-variant-calling-challenges" class="slide level2">
<h2>Somatic Variant Calling Challenges</h2>
<div class="columns">
<div class="column" style="width:55%;">
<p><strong>Germline calling assumes:</strong></p>
<ul>
<li>Heterozygous: ~50% VAF (typically 30-70% due to technical variation)</li>
<li>Homozygous: ~100% VAF</li>
</ul>
<p><strong>These assumptions fail in tumors:</strong></p>
<ul>
<li>Variable tumor purity (30-80%, which is effectively contamination by normal cells)</li>
<li>Clonal heterogeneity (subclones)</li>
<li>Copy number alterations</li>
<li><strong>VAF can be anywhere from &lt;1% to 100%</strong></li>
</ul>
<p><strong>Additional challenges:</strong></p>
<ul>
<li>Sequencing errors, alignment artifacts</li>
<li>FFPE artifacts
<ul>
<li>C&gt;T/G&gt;A deamination</li>
<li>8-oxoG which causes the G&gt;T transversions</li>
</ul></li>
<li>Sample contamination</li>
</ul>
<p><br> <img data-src="../assets/presentations/slidedeck_2/11_plot_VAF_distributions.png" data-position="center" alt="VAF distribution comparison"></p>
</div><div class="column" style="width:45%;">
<h4 id="lets-make-an-example-assuming-diploid-genome-and-cn2">Let’s make an example (assuming <strong>diploid genome</strong> and CN=2)</h4>
<p><strong>1. The Ideal Case</strong><br> <strong>Sample is 100% Tumor</strong></p>
<p>If a mutation is heterozygous (1 of 2 alleles): <span class="math display">\[\text{VAF} = \frac{1}{2} = \mathbf{50\%}\]</span></p>
<p><em>To a caller, this looks like a clear, standard germline variant.</em></p>
<p><strong>2. The Real World</strong><br> <strong>Sample is 40% Tumor</strong> (60% Normal)</p>
<p>The normal cells (wild type) dilute the signal. <span class="math display">\[\text{VAF} = \frac{\text{Purity}}{2}\]</span> <span class="math display">\[\text{VAF} = \frac{0.40}{2} = \mathbf{20\%}\]</span></p>
</div></div>
</section>
<section id="tumor-purity-ploidy-and-clonality" class="slide level2">
<h2>Tumor Purity, Ploidy, and Clonality</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Key concepts:</strong></p>
<p><strong>Tumor purity</strong>: Fraction of tumor cells in sample</p>
<p><strong>Ploidy</strong>: Average copy number across genome (often &gt;2 in cancer)</p>
<p><strong>Clonal variants</strong>: Present in all tumor cells</p>
<p><strong>Subclonal variants</strong>: Present in a subset of tumor cells</p>
<p><strong>VAF interpretation examples:</strong></p>
<ul>
<li>Pure tumor (100%), clonal het variant → VAF ≈ 50%</li>
<li>50% purity, clonal het variant → VAF ≈ 25%</li>
<li>50% purity, subclonal variant (20% of cells) → VAF ≈ 5%</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/12_clonal_evo.png" alt="Fish plot showing clonal evolution"></p>
<figcaption>Fish plot of clonal evolution: Panel a A case of primary and relapsed AML. Panel b A breast cancer before and after neoadjuvant aromatase inhibitor therapy. Panel c An AML with complex clonal structure and 7 timepoints, <a href="https://doi.org/10.1186/s12864-016-3195-z">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Clinical Relevance</strong></p>
</div>
<div class="callout-content">
<p>Subclonal variants can become dominant after treatment selection pressure, leading to resistance.</p>
</div>
</div>
</div>
</section>
<section id="variant-filtering-strategies" class="slide level2">
<h2>Variant Filtering Strategies</h2>
<p><strong>Three key considerations:</strong></p>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="sequencing-error">1. Sequencing Error</h3>
<ul>
<li>Base quality scores (Phred: Q30 = 1/1000 error)</li>
<li>Variant allele frequency</li>
<li>Strand bias</li>
<li>Mapping quality (ie MAPQ ≥ 20)</li>
</ul>
<h3 id="technical-artifacts">2. Technical Artifacts</h3>
<ul>
<li><strong>Panel of Normals (PoN)</strong>: Database of artifacts seen in normal samples</li>
<li>Systematic errors from library prep or sequencing</li>
<li>FFPE artifacts</li>
</ul>
</div><div class="column" style="width:50%;">
<h3 id="germline-filtering">3. Germline Filtering</h3>
<ul>
<li>Compare with matched normal</li>
<li>Filter using population databases:
<ul>
<li><strong>gnomAD v4.0</strong> (&gt;800K individuals)</li>
<li>1000 Genomes Phase 3 (~2,500 individuals)</li>
</ul></li>
<li>Common variants (AF &gt; 0.1%) are typically germline</li>
</ul>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>gnomAD</strong></p>
</div>
<div class="callout-content">
<p>The Genome Aggregation Database contains variant frequencies essential for filtering common germline variants. Most callers (Mutect2, Strelka2) apply these filters automatically, but understanding the logic helps with troubleshooting false positives.</p>
</div>
</div>
</div>
</section>
<section id="gatk-mutect2-workflow" class="slide level2">
<h2>GATK Mutect2 Workflow</h2>
<div class="cell" data-reveal="true" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class=""><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart TD
    A[Tumor BAM] --&gt; C[Mutect2]
    B[Normal BAM] --&gt; C
    D[Panel of Normals] --&gt; C
    E[Germline Resource] --&gt; C
    
    A -.-&gt; |optional| F[GetPileupSummaries]
    E -.-&gt; F
    F -.-&gt; G[CalculateContamination]
    G -.-&gt; |if available| H
    
    C --&gt; H[FilterMutectCalls]
    H --&gt; I[Filtered VCF]
    I --&gt; J[VEP Annotation]
    
    classDef resource fill:#e1f5ff,stroke:#0288d1,stroke-width:2px
    classDef process fill:#fff9c4,stroke:#f57f17,stroke-width:2px
    
    class D,E resource
    class C,F,G,H,J process
</pre>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p><strong>Key features:</strong></p>
<ul>
<li>Haplotype-aware variant calling (local assembly)</li>
<li>Joint analysis of tumor-normal pairs</li>
<li>Integrated contamination estimation</li>
<li>F1R2 (Forward 1st, Reverse 2nd) artifact filtering</li>
</ul>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>PoN vs gnomAD</strong></p>
</div>
<div class="callout-content">
<ul>
<li><strong>Germline Resource</strong>: Filters biological germline variants (universal)</li>
<li><strong>PoN</strong>: Filters technical artifacts (platform-specific)</li>
</ul>
<p>Public PoNs exist but work best when your sequencing protocol matches theirs. When in doubt, build your own!</p>
</div>
</div>
</div>
<p>You can find every step’s relative script <a href="https://github.com/sib-swiss/cancer-variants-training/tree/master/scripts">here</a></p>
</section>
<section id="the-vcf-file-format" class="slide level2">
<h2>The VCF File Format</h2>
<p><strong>Variant Call Format</strong> Standard for storing variant data (current: v4.3)</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="sourceCode numberSource vcf number-lines code-with-copy"><code class="sourceCode"><span id="cb3-1"><a></a>##fileformat=VCFv4.3</span>
<span id="cb3-2"><a></a>##INFO=&lt;ID=DP,Number=1,Type=Integer,Description="Total Depth"&gt;</span>
<span id="cb3-3"><a></a>##INFO=&lt;ID=AF,Number=A,Type=Float,Description="Allele Frequency"&gt;</span>
<span id="cb3-4"><a></a>##FILTER=&lt;ID=weak_evidence,Description="Insufficient support"&gt;</span>
<span id="cb3-5"><a></a>##FILTER=&lt;ID=germline,Description="Likely germline variant"&gt;</span>
<span id="cb3-6"><a></a>##FORMAT=&lt;ID=GT,Number=1,Type=String,Description="Genotype"&gt;</span>
<span id="cb3-7"><a></a>##FORMAT=&lt;ID=AD,Number=R,Type=Integer,Description="Allelic depths (REF,ALT)"&gt;</span>
<span id="cb3-8"><a></a>##FORMAT=&lt;ID=AF,Number=A,Type=Float,Description="Allele Frequency"&gt;</span>
<span id="cb3-9"><a></a>#CHROM  POS     ID        REF  ALT  QUAL  FILTER  INFO        FORMAT      TUMOR       NORMAL</span>
<span id="cb3-10"><a></a>chr17   7577538 rs123     G    A    .     PASS    DP=100      GT:AD:AF    0/1:70,30:0.30  0/0:50,0:0.0</span>
<span id="cb3-11"><a></a>chr17   7578406 .         C    T    .     germline DP=85     GT:AD:AF    0/1:40,45:0.53  0/1:30,25:0.45</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p><strong>Key columns explained:</strong></p>
<ul>
<li><strong>CHROM/POS</strong>: Genomic location</li>
<li><strong>REF/ALT</strong>: Reference and alternate alleles<br>
</li>
<li><strong>FILTER</strong>: Quality filters applied (PASS = passed all)</li>
<li><strong>FORMAT</strong>: Defines sample-specific fields</li>
<li><strong>GT</strong>: Genotype (0/0=hom ref, 0/1=het, 1/1=hom alt)</li>
<li><strong>AD</strong>: Allele depths (REF count, ALT count)</li>
<li><strong>AF</strong>: Variant Allele Frequency = ALT/(REF+ALT)</li>
</ul>
<p><strong>Example interpretation:</strong></p>
<p>Row 1: Somatic mutation (30% VAF in tumor, 0% in normal) ✓<br>
Row 2: Germline variant (present in both tumor and normal) ✗</p>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>FILTER Field</strong></p>
</div>
<div class="callout-content">
<p>“PASS” means the variant passed all filters, <strong>not</strong> that it’s biologically validated. Clinical decisions require orthogonal validation (Sanger, ddPCR, amplicon-seq).</p>
</div>
</div>
</div>
</section>
<section id="variant-annotation-the-critical-step" class="slide level2">
<h2>Variant Annotation: The Critical Step</h2>
<p>After calling, variants need biological context:</p>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="functional-annotation">Functional Annotation</h3>
<ul>
<li><strong>VEP</strong> (Ensembl) or <strong>SnpEff</strong></li>
<li>Effect prediction: missense, nonsense, splice site</li>
<li>Impact assessment: SIFT, PolyPhen, CADD scores</li>
</ul>
<h3 id="clinical-databases">Clinical Databases</h3>
<ul>
<li><strong>ClinVar</strong>: Clinical significance</li>
<li><strong>COSMIC</strong>: Cancer mutation database</li>
<li><strong>OncoKB</strong>, <strong>CIViC</strong>: Actionability</li>
</ul>
</div><div class="column" style="width:50%;">
<h3 id="population-frequencies">Population Frequencies</h3>
<ul>
<li>gnomAD, dbSNP</li>
<li>Filter common germline variants</li>
</ul>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tools</strong></p>
</div>
<div class="callout-content">
<p>VEP offers extensive plugin ecosystem.</p>
</div>
</div>
</div>
</section>
<section id="structural-variation-detection" class="slide level2">
<h2>Structural Variation Detection</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Types:</strong></p>
<ul>
<li>Large insertions/deletions</li>
<li>Translocations</li>
<li>Inversions</li>
<li>Complex rearrangements</li>
</ul>
<p><strong>Detection methods:</strong></p>
<ul>
<li>Discordant read pairs</li>
<li>Split reads</li>
<li>Read depth changes</li>
<li>Assembly-based approaches</li>
</ul>
<p><strong>Tools:</strong> Manta, Tiddit, GRIDSS2, DELLY, Sniffles2 (long reads)</p>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/13_SV.png" style="width:110.0%;height:110.0%" alt="IGV showing structural variant"></p>
<figcaption>SV: inversion in IGV, <a href="https://help.dragen.illumina.com/product-guide/dragen-v4.4/dragen-dna-pipeline/sv-calling/sv-igv-tutorial">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Long Reads Advantage</strong></p>
</div>
<div class="callout-content">
<p>Long-read sequencing (PacBio HiFi, ONT) dramatically improves structural variant detection, especially for complex rearrangements and insertions.</p>
</div>
</div>
</div>
</section>
<section id="copy-number-variation-cnv" class="slide level2 scrollable">
<h2>Copy Number Variation (CNV)</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Characteristics:</strong></p>
<ul>
<li>Gains or losses of genomic segments</li>
<li>Full chromosome or arm-level events common</li>
<li>Can cause Loss of Heterozygosity (LOH)</li>
</ul>
<p><strong>Detection approach:</strong></p>
<ol type="1">
<li>Calculate coverage in bins</li>
<li>Normalize for GC content and mappability</li>
<li>Compare tumor vs normal ratio</li>
<li>Segment and call CNV regions</li>
</ol>
<p><strong>Tools:</strong> CNVkit, ASCAT, Control-FREEC, PURPLE</p>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/14_CNV.png" style="width:110.0%;height:110.0%" alt="CNVKit scatter plot"></p>
<figcaption>CNV scatter pot, <a href="https://cnvkit.readthedocs.io/en/v0.7.6/plots.html?highlight=re">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Clinical Example</strong></p>
</div>
<div class="callout-content">
<p>ERBB2 (HER2) amplification in breast cancer determines eligibility for trastuzumab. MYC amplification is prognostic in many cancers.</p>
</div>
</div>
</div>
</section>
<section id="gene-fusions-in-cancer" class="slide level2">
<h2>Gene Fusions in Cancer</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Mechanism:</strong></p>
<ul>
<li>Chromosomal translocation</li>
<li>Fusion of gene elements</li>
<li>Creates chimeric transcripts</li>
</ul>
<p><strong>Detection data types:</strong></p>
<ul>
<li>WGS (genomic breakpoints)</li>
<li>WES (if breakpoints in exons)</li>
<li>RNA-seq (fusion transcripts)</li>
</ul>
<p><strong>Detection method:</strong> Discordant alignments where paired reads map to different genes/chromosomes</p>
<p><strong>Tools:</strong> Manta (DNA), STAR-Fusion, Arriba (RNA-seq)</p>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../assets/presentations/slidedeck_2/15_Gene_fusion.jpg" style="width:110.0%;height:110.0%" alt="CNVKit scatter plot"></p>
<figcaption>BCR::ABL1 fusion, <a href="https://www.cancer.gov/publications/dictionaries/cancer-terms/def/bcrabl1-fusion-gene">Source</a></figcaption>
</figure>
</div>
</div></div>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Famous Example</strong></p>
</div>
<div class="callout-content">
<p>BCR-ABL fusion in CML(Chronic Myeloid Leukemia)</p>
<ul>
<li>discovered 1960</li>
<li>translocation identified 1973</li>
<li>imatinib approved 2001.</li>
</ul>
<p>From observation to targeted therapy took 41 years; today we can do this computationally.</p>
</div>
</div>
</div>
</section>
<section id="quality-control-metrics" class="slide level2">
<h2>Quality Control Metrics</h2>
<p><strong>Essential QC checks at each stage:</strong></p>
<table class="caption-top">
<colgroup>
<col style="width: 20%">
<col style="width: 40%">
<col style="width: 40%">
</colgroup>
<thead>
<tr class="header">
<th>Metric</th>
<th>Expected Value</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Mapping rate</td>
<td>&gt;95%</td>
<td>Low = contamination or poor quality</td>
</tr>
<tr class="even">
<td>Duplicate rate</td>
<td>&lt;30% (WGS)</td>
<td>High = low library complexity</td>
</tr>
<tr class="odd">
<td>Mean coverage</td>
<td>As specified</td>
<td>Low = insufficient data</td>
</tr>
<tr class="even">
<td>Coverage uniformity</td>
<td>CV &lt;0.2</td>
<td>High variability = capture issues</td>
</tr>
<tr class="odd">
<td>Contamination</td>
<td>&lt;1-2%</td>
<td>High = sample swap or cross-contamination</td>
</tr>
<tr class="even">
<td>Ti/Tv ratio</td>
<td>~2.0-2.1 (WGS), ~3.0-3.3 (exome)</td>
<td>Low = sequencing errors enriched</td>
</tr>
<tr class="odd">
<td>Insert size</td>
<td>200-400bp (Illumina)</td>
<td>Bimodal = library issues</td>
</tr>
<tr class="even">
<td>GC bias</td>
<td>Flat across 30-70% GC</td>
<td>Strong bias = PCR or coverage issues</td>
</tr>
</tbody>
</table>
<p><strong>Tools:</strong> fastQC, MultiQC, mosdepth, GATK CollectHsMetrics</p>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Common Pitfall</strong></p>
</div>
<div class="callout-content">
<p>Always check for cross-sample contamination - rampant in shared facilities and has caused retractions!</p>
</div>
</div>
</div>
</section>
<section id="summary-and-key-takeaways" class="slide level2">
<h2>Summary and Key Takeaways</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Cancer Genomics Fundamentals:</strong></p>
<ul>
<li>Cancer is driven by genomic alterations</li>
<li>Both small variants and structural changes</li>
<li>Somatic vs germline distinction critical</li>
<li>~5-10% hereditary predisposition</li>
</ul>
<p><strong>Technical Considerations:</strong></p>
<ul>
<li>Tumor-normal paired design</li>
<li>Appropriate sequencing strategy</li>
<li>Quality preprocessing essential</li>
<li>Purity &gt;30% recommended</li>
</ul>
</div><div class="column" style="width:50%;">
<p><strong>Variant Calling:</strong></p>
<ul>
<li>Haplotype-aware methods (Mutect2)</li>
<li>Multiple filtering strategies</li>
<li>Standard file formats (BAM, VCF)</li>
<li><strong>Annotation is essential!</strong></li>
</ul>
<p><strong>Clinical Applications:</strong></p>
<ul>
<li>Mutational signatures reveal etiology</li>
<li>TMB predicts immunotherapy response</li>
<li>Fusions guide targeted therapy</li>
<li>CNV determines treatment options</li>
</ul>
</div></div>
</section>
<section id="references-and-resources" class="slide level2">
<h2>References and Resources</h2>
<p><strong>Key Papers:</strong></p>
<ul>
<li>Hanahan &amp; Weinberg (2011) Cell - Hallmarks of Cancer</li>
<li>Vogelstein et al.&nbsp;(2013) Science - Cancer Genome Landscapes</li>
<li>Alexandrov et al.&nbsp;(2020) Nature - Mutational Signatures</li>
<li>Cibulskis et al.&nbsp;(2013) Nature Biotech - MuTect</li>
<li>Karczewski et al.&nbsp;(2020) Nature - gnomAD</li>
</ul>
<p><strong>Databases:</strong></p>
<ul>
<li>COSMIC (cancer.sanger.ac.uk) - Somatic mutations</li>
<li>gnomAD (gnomad.broadinstitute.org) - Population frequencies</li>
<li>ClinVar - Clinical interpretations</li>
<li>OncoKB - Precision oncology knowledge</li>
</ul>
<p><strong>Tools &amp; Pipelines:</strong></p>
<ul>
<li>GATK (gatk.broadinstitute.org)</li>
<li>nf-core/sarek - Production pipeline</li>
<li>IGV (igv.org) - Visualization</li>
</ul>
</section>
<section id="exercises" class="slide level2" style="text-align: center">
<h2>Exercises</h2>



<img data-src="https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExOWdxc2J1OHY3eGs3c2ZycGhiNGt1dzQ4dmVlZXQxNjM2NDQxOWQybCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/zIOdLMZDcBDc2gk6vV/giphy.gif" class="r-stretch quarto-figure-center"><p class="caption">Giphy</p></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<p><img src="../assets/SIB_logo.svg" class="slide-logo"></p>
<div class="footer footer-default">
<p>Cancer Variant Analysis - SIB</p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': true,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'slide',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/sib-swiss\.github\.io\/cancer-variants-training\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>